pandas/tests/series/test_constructors.py

```diff
@@ -542,12 +542,29 @@ def test_constructor_pass_nan_nat(self):
         tm.assert_series_equal(Series(np.array([np.nan, pd.NaT])), exp)
 
     def test_constructor_cast(self):
-        pytest.raises(ValueError, Series, ['a', 'b', 'c'], dtype=float)
+        msg = "could not convert string to float"
+        with tm.assert_raises_regex(ValueError, msg):
+            Series(["a", "b", "c"], dtype=float)
+
```

Comment:
do we have these tests for Index as well (I think we do)?

Reply:
😮 : That's a bug!

~~~python
>>> Index([["a", "b", "c"], dtype=float)
Index([["a", "b", "c"], dtype=object)
~~~

Reply:
See #21311

Reply:
@jreback : I'll add an `xfail` test for this.